Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -133,48 +136,53 @@ private void parseStyleAddToMapStylesHolders()
         continue;
       }
       String type = mParser.getName();
-      if (type.equals("IconStyle")) {
-        parseIconStyle(stylesHolder);
-      } else if (type.equals("LineStyle")) {
-        parseLineStyle(stylesHolder);
-      } else {
-        skipToEndOfTag();
+      switch (type) {
+        case "IconStyle":
+          parseIconStyle(stylesHolder.getIconStyle());
+          break;
+        case "LineStyle":
+          parseLineStyle(stylesHolder.getLineStyle());
+          break;
+        case "PolyStyle":
+          parsePolyStyle(stylesHolder.getPolyStyle());
+          break;
+        default:
+          skipToEndOfTag();
+          break;
       }
     }
     mMapStylesHolders.put(id, stylesHolder);
   }
 
-  private void parseIconStyle(@NonNull StylesHolder stylesHolder)
+  private void parseIconStyle(@NonNull IconStyle iconStyle)
       throws XmlPullParserException, IOException, KMLParseException {
     while (moveToNext() != XmlPullParser.END_TAG) {
       if (mParser.getEventType() != XmlPullParser.START_TAG) {
         continue;
       }
-      String type = mParser.getName();
-      if (type.equals("Icon")) {
-        parseIcon(stylesHolder);
+      if (mParser.getName().equals("Icon")) {
+        parseIcon(iconStyle);
       } else {
         skipToEndOfTag();
       }
     }
   }
 
-  private void parseIcon(@NonNull StylesHolder stylesHolder)
+  private void parseIcon(@NonNull IconStyle iconStyle)
       throws XmlPullParserException, IOException, KMLParseException {
     while (moveToNext() != XmlPullParser.END_TAG) {
       if (mParser.getEventType() != XmlPullParser.START_TAG) {
         continue;
       }
-      String type = mParser.getName();
-      if (type.equals("href")) {
+      if (mParser.getName().equals("href")) {
         String url = parseText();
         InputStream inputStream = new URL(url).openConnection().getInputStream();
-        stylesHolder.getIconStyle().setImage(mFactory.createMapImage(inputStream));
+        iconStyle.setImage(mFactory.createMapImage(inputStream));
```

Comment:
The icon style mutating when parsing the icon seems wonky. Can you have two icons with the same style but with different hrefs on each icon?

Reply:
Resolved in Teams: Icon is an element under IconStyle. Icon may have an href element that describes the image to be used for the MapIcon. Point is an element under Placemark that describes the MapIcon for the map.